app/src/org/commcare/android/tasks/FormRecordCleanupTask.java

```diff
@@ -261,40 +275,26 @@ public void commit(String[] meta) throws IOException, SessionUnavailableExceptio
         } catch (InvalidKeyException e) {
             e.printStackTrace();
             throw new RuntimeException("Invalid Key Data while attempting to decode form submission for processing");
+        } finally {
+            fis.close();
+            if (is != null) {
+                is.close();
+            }
         }
 
-        //Construct parser for this form's internal data.
-        DataModelPullParser parser = new DataModelPullParser(is, factory);
-        parser.parse();
-
-        //TODO: We should be committing all changes to form record models via the ASW objects, not manually.
-        FormRecord parsed = new FormRecord(r.getInstanceURI().toString(), newStatus, r.getFormNamespace(), r.getAesKey(),uuid[0], modified[0]);
+        // TODO: We should be committing all changes to form record models via
+        // the ASW objects, not manually.
+        FormRecord parsed = new FormRecord(r.getInstanceURI().toString(),
+                newStatus, r.getFormNamespace(), r.getAesKey(),
+                uuid[0], modified[0]);
         parsed.setID(r.getID());
-        
-        //TODO: The platform adds a lot of unfortunate coupling here. Should split out the need to parse completely 
-        //uninitialized form records somewhere else.
-        
-        if(caseIDs[0] != null && r.getStatus().equals(FormRecord.STATUS_UNINDEXED)) {
-            AndroidSessionWrapper asw = AndroidSessionWrapper.mockEasiestRoute(platform, r.getFormNamespace(), caseIDs[0]);
-            asw.setFormRecordId(parsed.getID());
-            
-            SqlStorage<SessionStateDescriptor> ssdStorage = CommCareApplication._().getUserStorage(SessionStateDescriptor.class);
-            
-            //Also bad: this is not synchronous with the parsed record write
-            try {
-                ssdStorage.write(asw.getSessionStateDescriptor());
-            } catch (StorageFullException e) {
-            }
-        }
-        
-        //Make sure that the instance is no longer editable
-        if(!newStatus.equals(FormRecord.STATUS_INCOMPLETE) && !newStatus.equals(FormRecord.STATUS_UNSTARTED)) {
-            ContentValues cv = new ContentValues();

```

Comment:
STATUS_SAVED was always passed as the value for newStatus, so we always want to execute the body of this conditional.